controllers/spiaccesstoken_controller.go

```diff
@@ -166,21 +166,22 @@ func (r *SPIAccessTokenReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, nil
 	}
 
-	// cleanup tokens which are in awaiting state and have their parent binging removed.
-	// grace period taken in account to avoid race condition during the object creation process (i.e. when token is created but not yet linked)
-	if at.Status.Phase == api.SPIAccessTokenPhaseAwaitingTokenData && time.Since(at.CreationTimestamp.Time).Seconds() > NoLinkingBindingGracePeriodSeconds {
+	tokenLifetime := time.Since(at.CreationTimestamp.Time).Seconds()
+
+	// cleanup tokens by lifetime or on being unreferenced by any binding in the AwaitingToken state
+	if (tokenLifetime > r.Configuration.AccessTokenTtl.Seconds()) || (at.Status.Phase == api.SPIAccessTokenPhaseAwaitingTokenData && tokenLifetime > GracePeriodSeconds) {
 		hasLinkedBindings, err := hasLinkedBindings(ctx, &at, r.Client)
 		if err != nil {
-			lg.Error(err, "failed to validate the object", "token", at.ObjectMeta.Name, "error", err)
-			return ctrl.Result{}, fmt.Errorf("failed to validate the object: %w", err)
+			lg.Error(err, "failed to check linked bindings for token", "token", at.ObjectMeta.Name, "error", err)
+			return ctrl.Result{}, fmt.Errorf("failed to check linked bindings for token: %w", err)
 		}
 		if !hasLinkedBindings {
-			//if token is in Awaiting, and no linked bindings present, it means that it have no bindings referring to it and can be cleaned up
-			if err := r.Delete(ctx, &at); err != nil {
-				lg.Error(err, "failed to cleanup the processed token", "token", at.ObjectMeta.Name, "error", err)
-				return ctrl.Result{}, fmt.Errorf("failed to cleanup the processed token: %w", err)
+			err = r.Delete(ctx, &at)
+			if err != nil {
+				lg.Error(err, "failed to cleanup obsolete token", "binding", at.ObjectMeta.Name, "error", err)
```

Comment:
`binding` should be `token` but is unnecessary anyway, because the logger already has the reconciled token in its values.